src/main/scala/io/iohk/ethereum/consensus/pow/PoWMiningCoordinator.scala

```diff
@@ -0,0 +1,140 @@
+package io.iohk.ethereum.consensus.pow
+
+import akka.actor.typed.scaladsl.adapter._
+import akka.actor.typed.scaladsl.{AbstractBehavior, ActorContext, Behaviors}
+import akka.actor.typed.{Behavior, DispatcherSelector}
+import akka.actor.{ActorRef => ClassicActorRef}
+import akka.util.Timeout
+import io.iohk.ethereum.consensus.pow.PoWMiningCoordinator.CoordinatorProtocol
+import io.iohk.ethereum.consensus.pow.miners.{EthashDAGManager, EthashMiner, KeccakMiner, MinerProtocol}
+import io.iohk.ethereum.domain.{Block, Blockchain}
+import io.iohk.ethereum.jsonrpc.EthMiningService
+
+import scala.concurrent.duration.DurationInt
+import scala.util.Random
+
+object PoWMiningCoordinator {
+  // TODO in ETCM-773 make trait sealed
+  trait CoordinatorProtocol
+  final case class StartMining(mode: MiningMode) extends CoordinatorProtocol
+  case object DoMining extends CoordinatorProtocol
+  case object StopMining extends CoordinatorProtocol
+  case object MiningSuccessful extends CoordinatorProtocol
+  case object MiningUnsuccessful extends CoordinatorProtocol
+
+  // MiningMode will allow to remove MockerMiner
+  sealed trait MiningMode
+  case object RecurrentMining extends MiningMode // for normal mining
+  case object OnDemandMining extends MiningMode // for testing
+
+  sealed trait MiningResponse
+  case object MiningComplete extends MiningResponse
+
+  def apply(
+      syncController: ClassicActorRef,
+      ethMiningService: EthMiningService,
+      blockCreator: PoWBlockCreator,
+      blockchain: Blockchain,
+      ecip1049BlockNumber: Option[BigInt]
+  ): Behavior[CoordinatorProtocol] =
+    Behaviors.setup(context =>
+      new PoWMiningCoordinator(context, syncController, ethMiningService, blockCreator, blockchain, ecip1049BlockNumber)
+    )
+}
+
+class PoWMiningCoordinator private (
+    context: ActorContext[CoordinatorProtocol],
+    syncController: ClassicActorRef,
+    ethMiningService: EthMiningService,
+    blockCreator: PoWBlockCreator,
+    blockchain: Blockchain,
+    ecip1049BlockNumber: Option[BigInt]
+) extends AbstractBehavior[CoordinatorProtocol](context) {
+
+  import PoWMiningCoordinator._
+
+  private implicit val timeout: Timeout = 5.seconds
+  private val log = context.log
+  private val dagManager = new EthashDAGManager(blockCreator)
+
+  override def onMessage(msg: CoordinatorProtocol): Behavior[CoordinatorProtocol] = msg match {
+    case StartMining(mode) =>
+      log.info("Received message {}", StartMining(mode))
+      switchMiningMode(mode)
+  }
+
+  private def handleMiningRecurrent(): Behavior[CoordinatorProtocol] = Behaviors.receiveMessage {
+    case StartMining(mode) =>
+      log.info("Received message {}", StartMining(mode))
+      switchMiningMode(mode)
+
+    case DoMining =>
+      log.info("Received message ProcessMining")
+      blockchain
+        .getBestBlock()
+        .fold(
+          log.error("Unable to get block for mining: blockchain.getBestBlock() returned None")
```

Comment:
regarding the problems you're experiencing with pottery - might be that the startup process needs to be adjusted so that the client state is actually initialised to start mining.